resolver/ecs_resolver.go

```diff
@@ -0,0 +1,103 @@
+package resolver
+
+import (
+	"github.com/0xERR0R/blocky/config"
+	"github.com/0xERR0R/blocky/model"
+	"github.com/0xERR0R/blocky/util"
+	"github.com/miekg/dns"
+)
+
+const (
+	ecsSourceScope = uint8(0)
+	ecsIpv4Family  = uint16(1)
+	ecsIpv4Mask    = uint8(32)
+	ecsIpv6Family  = uint16(2)
+	ecsIpv6Mask    = uint8(128)
```

Comment:
Maybe we can rename these to look more enum like?
I didn't adjust any other code, probably easier to do with an IDE :stuck_out_tongue: 

Also changing the type of the masks make it possible to compare with the config values easily, and make sure we won't compare to the wrong config value.

```suggestion
// https://www.rfc-editor.org/rfc/rfc7871.html#section-6
const (
	ecsSourceScope = uint8(0)

	ecsMaskIPv4 = config.ECSv4Mask(net.IPv4len * 8)
	ecsMaskIPv6 = config.ECSv6Mask(net.IPv6len * 8)
)

// https://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml
const (
	ecsFamilyIPv4 = uint16(iota + 1)
	ecsFamilyIPv6
)
```